master/internal/plugin/oidc/service.go

```diff
@@ -272,19 +313,52 @@ func (s *Service) lookupUser(ctx context.Context, claimValue string) (*model.Use
 	return u, err
 }
 
+func mergeUserGroups(sessionData *IDTokenClaims, dbData *model.AgentUserGroup) *model.AgentUserGroup {
+	result := model.AgentUserGroup{
+		UID:   dbData.UID,
+		GID:   dbData.GID,
+		User:  dbData.User,
+		Group: dbData.Group,
+	}
+
+	if sessionData.AgentUIDSet {
```

Comment:
Another nit with this pattern.

I am imagining the following UX:

TL;DR: once a user as a UID/GID set, it can never be unset via Auth claims and would need to be manually removed.

1. System admin configures `security.default_task.user_id: 1234`, in `master.yaml 
2. Some impetus: results in `UserA` having AgentUID `5432` being set in the auth claim.
3. Other impetus: results in `UserA`'s AgentUID being removed from the auth claim _for which the admin would expect that tasks run by that user would run as UID 1234, however, I suspect they will continue to run as 5432._

To be clear, I am OK with this behavior, I think... but I wanted to bring it up here for the following reasons:
1. To get confirmation that I speculation about the behavior here is correct.
2. Document that this is the behavior we should expect.
3. Get any input from folks other than me if this behavior is acceptable.

Reply:
I tested this behavior today and confirmed that it works as I expected in might.

_Once you have enabled this feature and an AgentUID/GID is set for a user in order to have have this revert back to using the `security.default_task.uid` an admin needs to manually remove the AgentUID/GID settings for that user._

We discussed real would situations in which this type of thing could come and yesterday and the only real one we could think is if a customer switches IdPs and forgets to populate the Agent UID/GID attributes in the auth claim. In this situation we would likely want to keep the UID and GID of the users there and not remove them and have the system go back to user the default. Therefore, I think the current behavior is best.